src/types/AS5047P_Types.h

```diff
@@ -14,6 +14,9 @@
 
 #include <inttypes.h>
 
+//Following line needed for Teensy 4.1. Added by Ciarán McDermott
+#include <string>
```

Comment:
Hey Ciarán,
i appreciate your contribution, but your added line of code (`#include <string>`) here bricks the compatibility of this library with all Arduinos based on the Atmel AVR architecture (like the Arduino UNO). So I can't merge this change as it is right now. Don't get me wrong your solution will work in your case flawlessly but ist isn't universally applicable.

To give you some context if your using the default Teensy 4.1 toolchain (provided for example within the [PlatformIo](https://platformio.org/) VSCode extension) an identifier is defined called `CORE_TEENSY` when you compile projects for an Teenys board. If this identifier is present the following block of code ensures that the correct `string` library is included:

```c++
#if defined(ARDUINO_ARCH_SAMD) || defined(CORE_TEENSY)
#include <string>
#endif
```

It seams like for some reason this dosen't work in your use case. So I suspect that you are using some other toolchain to compile your code for your Teensy.
In order to fix this issue properly I would appreciate it if you could share what your current setup is which you use to compile your code. So than we will able to tweak the line `#if defined(ARDUINO_ARCH_SAMD) || defined(CORE_TEENSY)` and get it running for your setup as well as for all AVR base Arduino Boards. 

Best regards & happy coding
Jonas